lib/block.js

```diff
@@ -0,0 +1,147 @@
+'use strict';
+
+const stream = require('stream');
+const { PassThrough } = stream;
+const StringifierStream = require('./streams/stringifier-stream');
+const Fragment = require('./fragment');
+const FRAGMENT_EVENTS = ['start', 'response', 'end', 'error', 'timeout', 'fallback', 'warn'];
+
+// context -> quilt context -> fragment overrides
+// request -> original request info
+module.exports.buildBlock = function buildBlock(request, originalContext) {
+    // NOTE: should there be a separation between context as something global
+    // vs. context as quilt/tailor-specific term?
+    let context = originalContext;
+
+    const resultStream = new StringifierStream((tag) => {
+
+        const {
```

Comment:
this destructuring logic happens on every invocation of tag, should be moved out of of this function